appium/lib/screen-object/accessors/element.rb

```
@@ -81,43 +81,133 @@ def dynamic_text_exists? dynamic_text
         end
       end
 
-      def scroll
-        $driver.execute_script 'mobile: scrollTo',:element => element.ref
-        # $driver.execute_script("mobile: scroll",:direction => direction.downcase, :element => element.ref)
-      end
-
-      def scroll_to_text(text)
-        $driver.scroll_to(text)
+      def scroll(direction = 'down', touch_count = 1, duration = 1000)
+        # driver.execute_script 'mobile: scrollTo',:element => element.ref
+        # driver.execute_script("mobile: scroll",:direction => direction.downcase, :element => element.ref)
+        size = driver.window_size
+        x = size.width/2
+        y = size.height/2
+        if direction != 'up' && direction != 'down' && direction != 'left' && direction != 'right'
+          CXA.output_text 'Only upwards and downwards and leftwards and rightwards scrolling are supported for now'
```

Same as before, don't use `CXA` and you should actually raise an error here and not just simply output text.